internal/handlers/common/update_array_operators.go

```diff
@@ -374,34 +374,30 @@ func processPullAllArrayUpdateExpression(doc, update *types.Document) (bool, err
 			)
 		}
 
-		for i := 0; i < array.Len(); {
-			var value any
+		for j := 0; j < pullAllArray.Len(); j++ {
+			var valueToPull any
 
-			value, err = array.Get(i)
+			valueToPull, err = pullAllArray.Get(j)
 			if err != nil {
 				return false, lazyerrors.Error(err)
 			}
 
-			for j := 0; j < pullAllArray.Len(); j++ {
-				var valueToPull any
+			// we remove all instances of valueToPull in array
+			i := array.Len() - 1
+			for i >= 0 {
```

Comment:
```suggestion
			for i := array.Len() - 1; i >= 0 {
```

Reply:
Maybe even `for i := array.Len() - 1; i >= 0; i-- {`

Reply:
@rumyantseva I will make that change, it's more readable. Thanks.